posthog/hogql/bytecode.py

```diff
@@ -336,6 +337,90 @@ def visit_for_statement(self, node: ast.ForStatement):
             response.extend(self._end_scope())
         return response
 
+    def visit_for_in_statement(self, node: ast.ForInStatement):
+        response: list = []
+        self._start_scope()
+
+        key_var = node.keyVar
+        value_var = node.valueVar
+
+        # set up a bunch of temporary variables
+        expr_local = self._declare_local("__H_expr_H__")  # the obj/array itself
+        expr_keys_local = self._declare_local("__H_keys_H__")  # keys
+        expr_values_local = self._declare_local("__H_values_H__")  # values
+        loop_index_local = self._declare_local("__H_index_H__")  # 0
+        loop_limit_local = self._declare_local("__H_limit_H__")  # length of keys
+        key_var_local = self._declare_local(key_var) if key_var is not None else -1  # loop key
+        value_var_local = self._declare_local(value_var)  # loop value
+        response.extend([Operation.NULL] * (7 if key_var is not None else 6))
```

Comment:
7 else 6? Not  a biggy but I was like "wat?"

Reply:
Yeah, there's the `key_var` that sometimes gets declared as a local and sometimes is ommitted... 😅